solutions/gold/usaco-993.mdx

```diff
@@ -22,48 +32,54 @@ We only need to keep track of the number of days that Bessie has been traveling
 #include <bits/stdc++.h>
 using namespace std;
 
-const int MAX_DAYS = 1000;
-
 int main() {
-	freopen("time.in", "r", stdin);
-	freopen("time.out", "w", stdout);
+	ifstream cin("time.in");
+	ofstream cout("time.out");
 
 	int n, m, c;
 	cin >> n >> m >> c;
 
-	vector<int> earn(n);
-	for (int i = 0; i < n; i++) { cin >> earn[i]; }
+	vector<int> moonies(n + 1);
+	for (int i = 1; i <= n; i++) { cin >> moonies[i]; }
 
-	vector<vector<int>> adj(n);
+	vector<vector<int>> adj(n + 1);
 	for (int i = 0; i < m; i++) {
-		int u, v;
-		cin >> u >> v;
-		adj[--u].push_back(--v);
+		int a, b;
+		cin >> a >> b;
+		adj[a].push_back(b);
 	}
 
-	// dp[i][j] = the max money that Bessie can make on day i if she ends in
-	// city j
-	vector<vector<int>> dp(MAX_DAYS + 1, vector<int>(n, -1));
-	// base case: if Bessie doesn't travel at all, she makes $0
-	dp[0][0] = 0;
-
-	int ans = 0;
-	for (int d = 0; d < MAX_DAYS; d++) {
-		for (int i = 0; i < n; i++) {
-			// if dp[d][i] == -1 then the city can't be visited
-			if (dp[d][i] != -1) {
-				for (int u : adj[i]) {
-					/*
-					 * dp[d + 1][u] = max(current money earned,
-					 * previous city's earnings + current city's earnings)
-					 */
-					dp[d + 1][u] = max(dp[d + 1][u], dp[d][i] + earn[u]);
+	const int MAX_T = 1000;  // Maximum number of days Bessie can travel
+
+	// DP table: dp[t][i] represents the maximum moonies Bessie can have at city i on
+	// day t
+	vector<vector<int>> dp(MAX_T + 1, vector<int>(n + 1, -1));
```

Comment:
i would personally write

`dp[t][i] = max Moonies if at city i on day t`

to keep it a bit more concise